### PR TITLE
check the executor subject permission for delete

### PR DESCRIPTION
**Pull request by zhujian7**
issue reference: https://github.com/open-cluster-management-io/work/issues/154
Signed-off-by: zhujian <jiazhu@redhat.com>

### CODE REVIEW

File: pkg/spoke/controllers/manifestcontroller/manifestwork_controller.go [hunk 1]
```diff
@@ -274,15 +274,18 @@ func (m *ManifestWorkController) applyOneManifest(
 		return result
 	}
 
+	// check if the to be applied resource should be owned by the manifest work
```
Comment (qiujian16) [inline pkg/spoke/controllers/manifestcontroller/manifestwork_controller.go]: resource to be applied

File: pkg/spoke/auth/auth.go [hunk 2]
```diff
@@ -91,14 +81,9 @@ func (v *sarValidator) Validate(ctx context.Context, executor *workapiv1.Manifes
...
-	var verbs []string
-	switch action {
-	case ApplyAction:
-		verbs = []string{"create", "update", "patch", "get"}
-	case DeleteAction:
-		verbs = []string{"delete"}
-	default:
-		return fmt.Errorf("execute action %s is invalid", action)
+	verbs := []string{"create", "update", "patch", "get"}
+	if ownedByTheWork {
```
Comment (qiujian16) [inline pkg/spoke/auth/auth.go:85]: we should add a TODO here indicating the potential risk.

Comment (zhujian7) [reply pkg/spoke/auth/auth.go:85]: added some comments, ptal

**Pull request merged**